src/OrleansProviders/Streams/Common/PooledCache/IEvictionStrategy.cs

```diff
@@ -27,13 +23,24 @@ public interface IEvictionStrategy<TCachedMessage>
         /// </summary>
         /// <param name="utcNow"></param>
         /// <param name="purgeRequest"></param>
-        void PerformPurge(DateTime utcNow, IDisposable purgeRequest = null);
+        void PerformPurge(DateTime utcNow, FixedSizeBuffer purgeRequest = null);
 
         /// <summary>
         /// Method which should be called when data adapter allocated a new block
         /// </summary>
         /// <param name="newBlock"></param>
-        void OnBlockAllocated(IDisposable newBlock);
+        void OnBlockAllocated(FixedSizeBuffer newBlock);
+    }
+
+    public static class EvictionStrategyCommonUtils
+    {
+        public static void WireUpEvictionStrategy<TQueueMessage, TCachedMessage>(PooledQueueCache<TQueueMessage, TCachedMessage> cache,
```

Comment:
Util method to wire eviction strategy into cache system. it's really easy to forget those two little pieces. but without those, purging won't work !!